src/common/utils/http.ts

```diff
@@ -0,0 +1,44 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+export type httpResponse = request.Response;
+
+export async function httpGet(url: string, options?: request.CoreOptions): Promise<request.Response> {
+
+    const store = container.get("store") as Store<RootState>;
+    const locale = store.getState().i18n.locale;
+    const requestOptions = Object.assign(
+        {
+            url,
+            method: "GET",
+            encoding: undefined,
+            headers: {
+                "User-Agent": "readium-desktop",
+                "Accept-Language": `${locale},en-US;q=0.7,en;q=0.5`,
+            },
+        },
+        options,
```

Comment:
I've just realized: we must ensure that certain options cannot be overridden by the `options` function parameter.
1) `options.url` (otherwise, this creates confusion with the first function parameter `url`)
2) `options.method` (otherwise, the function cannot be called `httpGet()`)
3) `options.headers["User-Agent"]` (this is debatable, but I personally think API consumers of `httpGet()` should not be given a choice here)
4) `options.headers["Accept-Language"]` (same as above)
5) `options.encoding` (I propose that we guarantee that `httpGet()` returns `Promise<string>` (as per my other comment about `util.promisify`), so `encoding` should not be overridden by `httpGet()` callers)

Any other thoughts?
https://github.com/request/request#requestoptions-callback

Reply:
PS: I see two ways to prevent overriding:
1) `delete`:
```javascript
if (options) {
  if (typeof options.url !== "undefined") {
    delete options.url;
  }
  if (typeof options.encoding !== "undefined") { // note that this covers `null` too, as typeof null === "object"!
    delete options.encoding;
  }
}
```

2) `Object.assign()` override (place high-precedence object as last parameter)

Reply:
I will use the second method

Reply:
I think it is a better choice too, because `delete` would modify the passed object, which might break the API consumer / client side of `httpGet()`.

Reply:
done